File: app/src/main/java/com/liushu/crazyandroid/ui/stage01/chapter02/activity/Demo020504Activity.java
package com.liushu.crazyandroid.ui.stage01.chapter02.activity;

import android.widget.ImageView;
import android.widget.ListView;
import android.widget.SimpleAdapter;
import android.widget.TextView;

import com.jaydenxiao.common.base.BaseActivity;
import com.liushu.crazyandroid.R;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import butterknife.BindView;
import butterknife.OnClick;

public class Demo020504Activity extends BaseActivity {
    @BindView(R.id.iv_back)
    ImageView mIvBack;
    @BindView(R.id.tv_title_name)
    TextView mTvTitleName;
    @BindView(R.id.lv_test)
    ListView mLvTest;
    private String[] names = {"虎头", "弄玉", "李清照", "李白"};
    private String[] descs = {"一个可爱的女孩", "一个擅长音乐的女孩", "一个擅长文学的女性", "一个浪漫主义诗人"};
    private int[] mInts = {R.drawable.tiger, R.drawable.nongyu, R.drawable.qingzhao, R.drawable.libai};

    @Override
    public int getLayoutId() {
        return R.layout.activity_demo020504;
    }


    @Override
    public void initView() {
        mTvTitleName.setText("使用SimpleAdapter创建ListView");
        List<Map<String, Object>> listItems = new ArrayList<>();
        for (int i = 0; i < names.length; i++) {
            Map<String, Object> map = new HashMap<>();
            map.put("header", mInts[i]);
            map.put("personName", names[i]);
            map.put("descs", descs[i]);
            listItems.add(map);
        }
        SimpleAdapter simpleAdapter = new SimpleAdapter(this, listItems, R.layout.simple_item,
                new String[]{"header", "personName", "descs"}, new int[]{R.id.header, R.id.name, R.id.desc});
        mLvTest.setAdapter(simpleAdapter);
    }

    @OnClick(R.id.iv_back)
    public void onClick() {
        finish();
    }
}

File: app/src/main/java/com/liushu/crazyandroid/ui/stage01/chapter02/activity/Demo020504Activity.kt
package com.liushu.crazyandroid.ui.stage01.chapter02.activity

import android.widget.SimpleAdapter
import butterknife.OnClick
import com.jaydenxiao.common.base.BaseActivity
import com.liushu.crazyandroid.R
import kotlinx.android.synthetic.main.activity_demo020504.*
import kotlinx.android.synthetic.main.head_title_layout.*
import java.util.*

class Demo020504Activity : BaseActivity() {
    private val names = arrayOf("虎头", "弄玉", "李清照", "李白")
    private val descs = arrayOf("一个可爱的女孩", "一个擅长音乐的女孩", "一个擅长文学的女性", "一个浪漫主义诗人")
    private val mInts = intArrayOf(R.drawable.tiger, R.drawable.nongyu, R.drawable.qingzhao, R.drawable.libai)

    override val layoutId: Int
        get() = R.layout.activity_demo020504


    override fun initView() {
        tv_title_name!!.text = "使用SimpleAdapter创建ListView"
        val listItems = ArrayList<Map<String, Any>>()
        for (i in names.indices) {
            val map = HashMap<String, Any>()
            map["header"] = mInts[i]
            map["personName"] = names[i]
            map["descs"] = descs[i]
            listItems.add(map)
        }
        val simpleAdapter = SimpleAdapter(this, listItems, R.layout.simple_item,
                arrayOf("header", "personName", "descs"), intArrayOf(R.id.header, R.id.name, R.id.desc))
        lv_test!!.adapter = simpleAdapter
    }

    @OnClick(R.id.iv_back)
    fun onClick() {
        finish()
    }
}
